### PR TITLE
Make AuthConfig fluent, setters -> with.

### LINKED ISSUE #453: non-void setters for AuthConfig
Hope class shouldn't match java beans and can provide fluent setters.

### PR DESCRIPTION
**Pull request by KostyaSha**
Resolves #453

### CODE REVIEW

**Comment by KostyaSha** [comment]
I failed checkstyle :)

File: src/main/java/com/github/dockerjava/api/model/AuthConfig.java [hunk 1]
```diff
@@ -30,48 +29,52 @@
...
         this.email = email;
+        return this;
     }
 
     public String getServerAddress() {
         return serverAddress;
     }
 
-    public void setServerAddress(String serverAddress) {
+    public AuthConfig withServerAddress(String serverAddress) {
```
Comment (marcuslinke) [inline src/main/java/com/github/dockerjava/api/model/AuthConfig.java]: Not sure if this should be included in this PR but I suggest to rename method to `withRegistryUrl`. WDYT?

Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/model/AuthConfig.java]: I think it not url, it just addr:port. Probably `withRegistryAddr` is better

**Comment by KostyaSha** [comment]
@marcuslinke added as separate commit

File: src/main/java/com/github/dockerjava/api/model/AuthConfig.java [hunk 2]
```diff
@@ -30,48 +29,52 @@
...
         this.email = email;
+        return this;
     }
 
     public String getServerAddress() {
         return serverAddress;
     }
 
-    public void setServerAddress(String serverAddress) {
+    public AuthConfig withRegistryAddres(String serverAddress) {
```
Comment (marcuslinke) [inline src/main/java/com/github/dockerjava/api/model/AuthConfig.java]: found typo :)

Comment (marcuslinke) [inline src/main/java/com/github/dockerjava/api/model/AuthConfig.java]: rename argument also please

**Comment by KostyaSha** [comment]
:E done

**Comment by marcuslinke** [comment]
LGTM. Feel free to merge!

**Pull request merged**